src/app/map/components/GoogleMaps/index.tsx

```diff
@@ -148,9 +154,22 @@ export const GoogleMaps = ({ setMapLoaded }: GoogleMapsProps) => {
         );
       });
       setMarkers(markerElements as JSX.Element[]);
+    }else {
+        setEvents(false);
+        setNoEvents(true);
+      }
     }
   }, [loading, error, result, infoWindowID]);
 
+    const [noEvents, setNoEvents] = useState(false);
+    const [events, setEvents] = useState(false);
+
```

Comment:
@SachiGoto I feel like we can check if there are events or not with only one boolean state value.

Reply:
ohhh wait a minute, that's true!! I will refactor the code!

Reply:
Just refactored the code. It has smooth transition of popup messages too!

Reply:
Nicely done!! 🙌